bindings/go/opendal_test.go

```diff
@@ -30,16 +30,14 @@ import (
 	"sync"
 	"testing"
 
+	"github.com/apache/opendal-go-services/memory"
```

Comment:
Maybe we can move those tests to another package?

Reply:
No problem.

Reply:
> No problem. What's your advice? In fact, it is already another package:

I want to ensure that `github.com/apache/opendal-go-services` does not appear in our `go.mod` or `go.sum`, as it could impact our release.

Reply:
Ok.  Where do you think I should place those tests?

Reply:
Is it valid to have `bindings/go/tests/go.mod`? Will `bindings/go/go.mod` still contains the dependences of `tests`?

Reply:
Yes, it's valid. And `bindings/go/go.mod` will not contain dependencies of tests. But the release may still contain a tests folder. I'm not sure.

Reply:
> But the release may still contain a tests folder. I'm not sure.

We can exclude the `tests` folder entirely.

Reply:
IMO it is better to have a go workspace or something like that. Each of them doesn't contain others. And the developer of OpenDAL can have good experience in testing it.

Reply:
I'll resolve this latter.